File: test_data/code/java/1710-maximum-units-on-a-truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a, b) -> b[1] - a[1]);
        int totalUnits = 0;
        
        for (int[] boxes : boxTypes) {
            int boxesToTake = Math.min(truckSize, boxes[0]);
            totalUnits += boxesToTake * boxes[1];
            truckSize -= boxesToTake;
            if (truckSize == 0) break;
        }
        
        return totalUnits;
    }
}

File: test_data/code/python/1710-maximum-units-on-a-truck.py
from typing import List

class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x: x[1], reverse=True)
        totalUnits = 0
        for numBoxes, numUnits in boxTypes:
            if truckSize >= numBoxes:
                totalUnits += numBoxes * numUnits
                truckSize -= numBoxes
            else:
                totalUnits += truckSize * numUnits
                truckSize = 0
                break
        return totalUnits